sdk/CONTRIBUTING.md

```diff
@@ -1,12 +1,69 @@
 ## Contributing to the `kfp` SDK
-KFP v2 for SDK is under development on the `master` branch.
 
-To contribute to v1, please go to [sdk/release-1.8 branch](https://github.com/kubeflow/pipelines/tree/sdk/release-1.8) to submit your change.
+For developing KFP v2 for SDK use the `master` branch.
+
+For developing KFP v1 for SDK use the [sdk/release-1.8 branch](https://github.com/kubeflow/pipelines/tree/sdk/release-1.8) to submit your changes.
 
 For general contribution guidelines including pull request conventions, see [pipelines/CONTRIBUTING.md](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md).
 
-### Requirements
-All development requirement versions are pinned in [requirements-dev.txt](https://github.com/kubeflow/pipelines/blob/master/sdk/python/requirements-dev.txt).
+### Pre-requisites 
+
+Clone the repo: 
+
+```bash
+git clone https://github.com/kubeflow/pipelines.git && cd pipelines
+```
+
+We recommend working within a [virtual env](https://docs.python.org/3/library/venv.html) when developing for the SDK.
+
+```bash
+python -m venv env && source ./env/bin/activate
+```
+
+As with any Python package development, first ensure [wheels and setuptools](https://realpython.com/python-wheels/) are installed:
+
+```bash
+python -m pip install -U pip wheel setuptools
+```
+
+All sdk development requirement versions are pinned in [requirements-dev.txt](https://github.com/kubeflow/pipelines/blob/master/sdk/python/requirements-dev.txt).
+
+```bash
+pip install -r sdk/python/requirements-dev.txt
+```
+
+Install the SDK in development mode using the `--editable` or `-e` flag. This will allow you to implement and test changes iteratively.
+Read more about it [here](https://setuptools.pypa.io/en/latest/userguide/development_mode.html).
+
+```bash
+pip install -e sdk/python
+```
+
+The SDK also relies on a couple other python packages also found within KFP.
+These consists of the [api proto package](https://github.com/kubeflow/pipelines/tree/master/api) and the kfp [kubernetes_platform](https://github.com/kubeflow/pipelines/tree/master/kubernetes_platform) package.
+
+For the proto code, we need protobuf-compiler to generate the python code. Find instructions for your system for your package manager [here](https://grpc.io/docs/protoc-installation/).
+For Linux, using apt or apt-get, for example:
+```bash
+apt-get install -y protobuf-compiler
```

Comment:
latest version of protoc causes problems. I've found that 21.12 works. I made a note about it in kubernetes_platform (https://github.com/kubeflow/pipelines/tree/master/kubernetes_platform#dependencies) but missed adding a similar note here.

Related, I guess we should share the changes in this PR with kubernetes_platform somehow.

Reply:
I've just went ahead and linked to that section instead